### ISSUE TITLE
RFC: remove theming

### ISSUE THREAD

**Issue by tkh44** [tkh44/emotion#212]
I'm starting to think we should remove built in theming and just let users use the `theming` library themselves. Having theming built in adds overhead and code complexity to `styled` that I don't think is necessary. I prefer to just import my theme file but I may be an outlier.

Some notes.
If we remove theming...

- `Styled` becomes a pure functional component and we can create a factory with [createEagerFactory](https://github.com/acdlite/recompose/blob/master/docs/API.md#createeagerfactory) to call it 🔥
- Bundle size will drop
- Performance will be better
- Users that want theming can just wrap styled with withTheme
- We **may** be able to add theming conditionally with babel option (we could just wrap styled with `withTheme` and import it if the user wants theming)

**Comment by Connorelsea** [comment]
I'd be happy with the decrease in bundle size when not using theming. Emotion currently sitting around 3.4kb in my app

![image](https://user-images.githubusercontent.com/1844458/28756157-7f487694-752e-11e7-92e0-075ac92bee1c.png)

**Comment by thangngoc89** [comment]
@Connorelsea since version 6.0 , you have to opt-in for themeing to use it.

**Comment by tkh44** [comment]
closed in #225

**Comment by ericclemmons** [comment]
Personally, I wouldn't mind if `theming` could be automatically like so:

```js
import styled from "emotion/react/themed";

export default styled.div`...`;
```

Which would be the equivalent of:

```js
import styled from "emotion/react";
import { withTheme } from "theming";

export default withTheme(styled.div`...`);
```

Having converted a large `styled-components` repo to `emotion`, everything was largely find/replace _besides_ wrapping all of the components with `withTheme`.
